### ISSUE TITLE
Define channel ranges in filters so Channel IDs can be sorted by Group.

### ISSUE THREAD

**Issue by ajmmm** [xteve-project/xTeVe#148]
I tend to filter by Group Name, because then channels are logically grouped. However, channel mapping is just sequential. So if a new channel is added to a group, or I change filters, I cannot easily re-organise channel mapping.

It would be awesome if channel numbers could be mapped sequentially by group, and if new channels are inserted into groups, channel numbers are sequentially reset. Alternatively, perhaps filters can be defined with channel number ranges, so the user can define these.
For example, Filter 1 could have channel range 1000 - 1499, Filter 2 could have channel range 1500 - 1999, and Filter 3 could have channel range 2000 - 2499. As channels are added or removed, they would get inserted into the ranges specified, so that channels remain sequentially sorted by filters.

Alternatively perhaps it would be easier to simply add a 'reset channel numbers' such that xTeVe just resets channel numbers in the order they're filtered.

**Comment by mar-mei** [comment]
The filter does not replace the mapping. The filter is used to reduce playlists with several 1000 streams to a few hundred.
Your idea would not work anymore if the channel number was changed in the mapping or if the specified range was exceeded by adding new channels. Newly added channels must also be activated manually in Plex.

xTeVe cannot do everything automatically for everyone, it should only allow you to have an individual, identical playlist for all compatible clients (Plex, Emby, M3U).

**Comment by ajmmm** [comment]
Yep, but of those few 100 a filter lets you get it down to, it doesn't seem possible to sort channel numbers easily. At least, not unless I'm missing something obvious.

I agree the suggestion isn't perfect, but it's what sprung into my mind when I was looking at the UI. My brain wanted to specify ranges for each mapped filter. If there's a more practical way that would also be cool. Right now, if I have channel 110 - 199 defined, and I want to add a channel in at 115, I have to shuffle every channel from 115-119 down manually. If I simply try to override 115, it puts the new channel down to the next available channel number.

Also agree xTeVe can't do everything automatically for everyone, but given xTeVe is the caninical source of truth for the channel number, and given it's where the channel filtering and mapping occurs, would you agree it does make sense to be able to apply the channel ordering here, such that it does apply to all clients sourcing data from xTeVe?

Thank you for any and all support and effort, xTeVe is brilliant.

**Comment by jaylex32** [comment]
I suggest this a year ago but it was never implemented. The developer doesn't have the interest in fixing the mapping settings or add any new features on xteve. What you see is what you get. Hopefully one day he will change he's opinion about it.

**Comment by stevehop** [comment]
I also suggested this (see issue #112 )
